enclave/src/accounts_nonce_storage/mod.rs

```diff
@@ -197,13 +197,17 @@ impl AccountsNonceStorage {
 
     // Nonce related functions
 
+    fn get_nonce(&self, acc: AccountId) -> Result<u32, AccountRegistryError> {
+        Ok(self.nonce_storage.read_nonce(acc)?)
```

Comment:
A little bit nit picking but this is also a time issue: You're unwrapping here just to wrap it again.
I think `self.nonce_storage.read_nonce(acc)` should suffice.